src/sparc/client/client.py

```diff
@@ -27,15 +57,37 @@ def __init__(self, config_file="config/config.ini", connect=True) -> None:
             logging.debug(str(config))
         current_config = config["global"]["default_profile"]
 
-        logging.debug(str(current_config))
+        # logging.debug(str(current_config))
+        logging.debug("Using the following config")
+        logging.debug(str(config[current_config]))
 
         # iterate through the modules in the current package
         package_dir = os.path.join(Path(__file__).resolve().parent, "services")
 
-        for (_, module_name, _) in iter_modules([package_dir]):
-
+        for _, module_name, _ in iter_modules([package_dir]):
             # import the module and iterate through its attributes
-            module = import_module(f"{__package__}.services.{module_name}")
+            self.add_module(
+                f"{__package__}.services.{module_name}", config[current_config], connect
+            )
+
+    def add_module(self, path, config, connect=True):
+        """Adds and optionally connects to a module in a given path with configuration variables defined in config.
+
+        Parameters
+        ----------
+        path : str
+            a path to the module
+        config : dict or configparser.SectionProxy
+            a dictionary (or Section of the config file parsed by ConfigParser) with the configuration variables
+        connect : bool
+            determines if the module should auto-connect
+        """
+        module_name = path.split(".")[-1]
+        try:
+            module = import_module(path)
+        except ImportError:
+            print(sys.exc_info())
```

Comment:
why using ``logging`` for some things and ``print`` for others? I suggest using ``logging`` for everything and avoid ``print`` . For instance, here I would suggest
```python
try:
     module = import_module(path)
except ImportError:
    logging.debug("Skipping module. Failed to import from %s", f"{path=}", exc_info=True)
```